### ISSUE TITLE
Should Reinforcement Pad drop units of similar value for each faction?

### ISSUE THREAD

**Issue by xezon** [TheSuperHackers/GeneralsGamePatch#1562]
Should Reinforcement Pad drop units of similar value for each faction?

Currently it drops Assault Troopcrawler for Infantry, which is much more value than a Crusader Tank.

### Proposal 1

Drop units of similar value in Reinforcement Pad.

### Proposal 2

Lengthen drop interval for high value loads ~if possible~. Is possible.

### Proposal 3

First drop immediately after capture.

**Comment by ExiLe-ZH** [comment]
| Faction | Payload               | BuildCost | BuildTime |
|---------|-----------------------|----------:|----------:|
| USA     | Crusader Tank         | 900       | 10        |
| SWG     | Tomahawk              | 1400      | 20        |
| AFG     | Humvee                | 800       | 10        |
| Laser   | Laser Tank            | 900       | 10        |
| China   | Battle Master         | 800       | 10        |
| Tank    | Battle Master         | 700       | 12        |
| Nuke    | Battle Master         | 800       | 12        |
| Inf     | Assault Troop Crawler | 2400      | 25        |
| GLA     | Scorpion Tank         | 600       | 7         |
| Tox     | Toxin Tractor         | 650       | 5         |
| Demo    | 2 Demo Bikes          | 1100      | 10        |
| Stealth | Quad Cannon           | 700       | 6         |

**Comment by Stubbjax** [comment]
Change Assault Troop Crawler to Dragon Tank and it'd be good. Maybe vChina could do with a Gattling Tank instead of a Battlemaster as well.

**Comment by xezon** [comment]
Can you also add the unit values to the info table?

**Comment by ExiLe-ZH** [comment]
would be cool if the overall value of the drop would be higher, like 2 paladins, 2 tomahawks to match the troopcrawler.

**Comment by xezon** [comment]
Jundiyy:
> Give all factions a unit of equal value which all generals have in common. 
> all USA: ambulance
> all GLA: quad
> all China: dragon tank

> I do like but not sure as 2 of them are attacking units. 
Pad could be made more useful tbh, it has low hp and gives very little for most units and have to wait 2 mins.

> Perhaps it could drop off every min or 1 min 30

> Or drop off 2 units at a time.
> Oil is $2000 for 2 mins and Pad is around $800 per 2 mins. Taking out the ATC.

**Comment by xezon** [comment]
Not dropping the Assault Troop Crawler would also prevent this issue:
* TheSuperHackers/GeneralsGameCode#2741

**Comment by xezon** [comment]
Drop Table updated. The SWG Tomahawk is also too much.

**Comment by xezon** [comment]
How about this? Changes in bold.

| Faction | Payload               | BuildCost | BuildTime |
|---------|-----------------------|----------:|----------:|
| USA     | **Paladin Tank**      | 1100      | 12        |
| SWG     | Tomahawk              | 1400      | 20        |
| AFG     | Humvee                | 800       | 10        |
| Laser   | Laser Tank            | 900       | 10        |
| China   | **2 Battle Master**   | 1600      | 20        |
| Tank    | Battle Master         | 700       | 12        |
| Nuke    | Battle Master         | 800       | 12        |
| Inf     | **Assault Outpost**   | 1000      | 15        |
| GLA     | Scorpion Tank         | 600       | 7         |
| Tox     | Toxin Tractor         | 650       | 5         |
| Demo    | 2 Demo Bikes          | 1100      | 10        |
| Stealth | Quad Cannon           | 700       | 6         |

**Comment by ExiLe-ZH** [comment]
Faction | Payload | BuildCost | BuildTime
-- | -- | -- | --
USA | Paladin Tank | 1100 | 15.3
SWG | Tomahawk | 1400 | 23.3
AFG | Humvee | 800 | 13.3
Laser | Laser Tank | 900 | 13.3
China | 2 Battle Master | 1600 | 26.6
Tank | Battle Master | 700 | 15.3
Nuke | Battle Master | 800 | 15.3
Inf | Assault Outpost | 1000 | 18.3
GLA | Scorpion Tank | 600 | 16
Tox | Toxin Tractor | 650 | 12
Demo | 2 Demo Bikes | 1100 | 24
Stealth | Quad Cannon | 700 | 14

This is with correct(?) patch build times.

**Comment by penfriendz** [comment]
Instead of making the reinforcement pad better than nothing, why don't we make it "good"?
USA: Crusader, Humvee
SWG: Humvee, ambo
AFG: Humvee, ambo
Laser: Laser tank, Humvee
China: Battlemaster x2, gattling
Tank: Battlemaster, gattling
Nuke: Battlemaster, gattling
Inf: Outpost, dragon (could swap dragon for two minigunners?)
Gla: scorpion, quad
Tox: scorpion, quad (who wants a toxin tractor, really)
Demo: scorpion, quad (two dbs is ok too, but I like the consistency and it also stops pro rules players complaining about DBs)
Stealth: quad x2

**Comment by ExiLe-ZH** [comment]
### Few suggestions.

### 1. Main Battle Unit:

Faction | Payload | BuildCost | BuildTime
-- | -- | -- | --
USA | Paladin Tank | 1100 | 15.3
SWG | Tomahawk | 1400 | 23.3
AFG | Humvee | 800 | 13.3
Laser | Laser Tank | 900 | 13.3
China | Listening Outpost | 800 | 18.3
Tank | Battle Master | 700 | 15.3
Nuke | Battle Master | 800 | 15.3
Inf | Assault Outpost (2 TH) | 1000 | 18.3
GLA | Marauder | 800 | 22
Tox | Marauder | 900 | 22
Demo | Marauder | 900 | 22
Stealth | Quad | 700 | 14

### 2. Double Main Battle Unit:

Faction | Payload | BuildCost | BuildTime
-- | -- | -- | --
USA | 2x Paladin | 2200 | 30.6
SWG | 2x Tomahawk | 2800 | 46.6
AFG | 2x Humvee | 1600 | 26.6
Laser | 2x Laser Tank | 1800 | 26.6
China | 2x Listening Outpost | 1600 | 36.6
Tank | 2x Battle Master | 1400 | 30.6
Nuke | 2x Battle Master | 1600 | 30.6
Inf | Assault Troop Crawler (5x MG) | 2400 | 28.3
GLA | 2x Marauder | 1600 | 44
Tox | 2x Marauder | 1800 | 44
Demo | 2x Marauder | 1800 | 44
Stealth | 2x Quad | 1400 | 28

### 3. High Value Payload:

Faction | Payload | BuildCost | BuildTime
-- | -- | -- | --
USA | Tomahawk | 1200 | 23.3
SWG | Tomahawk | 1400 | 23.3
AFG | Tomahawk | 1300 | 23.3
Laser | Avenger | 1500 | 13.3
China | Troop Crawler | 1400 | 18.3
Tank | Troop Crawler | 1400 | 18.3
Nuke | Troop Crawler | 1400 | 18.3
Inf | Assault Troop Crawler (4x MG) | 2400 | 28.3
GLA | Scud Launcher | 1200 | 42
Tox | Battle Bus | 1100 | 32
Demo | Bomb Truck | 1000 | 32
Stealth | Rocket Buggy | 900 | 22

**Comment by MTKing4** [comment]
I think we should minimize the value difference as much as possible, with maximum difference of $200 Preferably

**Comment by ExiLe-ZH** [comment]
Depends on how well the values reflect the effectiveness of the unit. These might also need some finetuning in this patch.

**Comment by MTKing4** [comment]
This one is good, with a few tweaks: 

### 1. Main Battle Unit:

Faction | Payload | BuildCost | BuildTime
-- | -- | -- | --
USA | Crusader Tank | 900 | 13.3
SWG | Sentry Drone | 1000 | 13.3
AFG | Humvee | 800 | 13.3
Laser | Laser Tank | 900 | 13.3
China | Listening Outpost | 800 | 18.3
Tank | Battle Master | 700 | 15.3
Nuke | Battle Master | 800 | 15.3
Inf | Assault Outpost (2 TH) | 1000 | 18.3
GLA | Marauder | 800 | 22
Tox | Scorpion Tank | 650 | 16
Demo | 2 Demo Bikes | 1100 | 24
Stealth | Quad | 700 | 14

(?) Might need BT and cost correction, feel free to do so

I don't like having different factions dropping the same units, makes it a bit boring so i went ahead and replaced the maurader for tox and demo among other changes

**Comment by MTKing4** [comment]
I've changed sentry drone count from 2 to 1 for swg, didn't know sentry drones were that expensive geez 😄

**Comment by ExiLe-ZH** [comment]
That was the point I was trying to make, Sentry Drones suck and aren't worth the price, so two was fair. I think we made them cheaper though?

**Comment by MTKing4** [comment]
yeah i don't remember how much we changed them though, it depends on the new price and buffs i guess

**Comment by ExiLe-ZH** [comment]
Regardless, 1 is just useless. Even 2 would be useless if you have to purchase the mg yourself.

Anyways, I like the idea of variety. Makes it a bit harder to balance though.

**Comment by MTKing4** [comment]
they can be used for scouting still, but yeah 1 is not enough, all other armies have better damage dealers so 2 sounds better

**Comment by kABUSE1** [comment]
I see a multitude of problems with the original reinforcement pad and some suggestions made here.

- should not drop stuff that is potentially not even unlocked (Marauders / Tomahawks)
- should not drop stuff that is not allowed as per "pro rules" (demo bikes)
- should not drop stuff that can make you passively go out-of-power (laser tanks)
- should not drop different stuff for each general, as that's just counter-intuitive and makes it hard to balance, as none of the prices match and usefulness would just be entirely random. I'd just change it to quads, humvees and outposts. Available to all generals, all can attack air, all within comparable price-range and build-time. I think that's good.

But none of it matters if you don't **_massively_** buff the drop rate first. As it is, nobody bothers capturing it and even if, it rarely ever stays alive long enough to do a noteworthy amount of drops.

Let's do a little math:
An oil derrick generates $200 per 12 seconds, so $2.000 in the pad drop delay of 2min. Also an oil derrick has no initial wait time but an initial cash buff instead. So the thought of a building dropping a single $650-$1100 vehicle after 2 minutes (plus additional time it takes for the plane to travel) is just ridiculous.

My suggestion would be to instantly trigger the first drop once captured (this has the same feel as the intial $1000 you get from a derrick, so you try extra hard to stop your enemy from capturing) and then re-drop once a minute.

Please take into account how reinforcement pads are almost always somewhere in the middle of the map, so very hard to defend and the health of them is very, very low, making it hard to keep them alive.

**Comment by xezon** [comment]
Very good points. I am not against making the Pad more valuable.

**Comment by MTKing4** [comment]
while having the same units makes it more balanced, it will be very boring.

**Comment by kABUSE1** [comment]
> while having the same units makes it more balanced, it will be very boring.

I mean,  yes, that's true.

But my suggestion to make it instantly trigger the first drop once captured alone already makes it less boring than it currently is, even when combined with "same stuff for all sub-factions".

I don't think imbalanced drops (demo bikes, tomahawks etc.) and insta-drop can co-exist.

I personally would go for the insta-drop variant, as realistically speaking you will virtually never be able to use reinforcement pads in a meaningful way if the first drop takes ages to arrive.

**Comment by ExiLe-ZH** [comment]
I agree same units is kinda boring, but on the other hand it's hard to balance all unique units.

**Comment by GoodGh0st** [comment]
Can drop times be changed for each general? That way we can give them same payload and have "pad is more useful for gen A than gen B"

**Comment by xezon** [comment]
> Can drop times be changed for each general? That way we can give them same payload and have "pad is more useful for gen A than gen B"

Good idea. The OCL's are indeed separate for each unit with `ObjectCreationList OCL_ReinforcementPad*`.

They can have unique `DropDelay` values.

**Comment by AIR-Z** [comment]
I think each general should have a choice of which unit to order next from a possible list. For example, we're playing as a toxin general, and he has the following options: Scorpion, Tractor, BombTruck, Quad—these are options that the player can switch to the next drop unit during gameplay. Each choice resets the reset timer. Each unit has its own reset interval.

Why am I presenting it this way in the conversation? The main unique problems we have with the infantry general are due to his colossal unit cost per unit of time. Let's give him the choice of sending a dragon faster, or an assault armored personnel carrier that takes a little longer—it's up to the player to decide what they need.
